pkg/controller/application/reconciliation/application_reconciler.go

```diff
@@ -107,28 +144,93 @@ func (a *applicationReconciler) hasCreateDependenciesMet(meta reconciliation.Met
 		return false, fmt.Errorf("determining existence of a Github repository for %s: %w", a.String(), err)
 	}
 
-	if meta.ApplicationDeclaration.Image.HasName() {
-		exists, err := state.ContainerRepository.ApplicationHasImage(meta.ApplicationDeclaration.Metadata.Name)
-		if err != nil {
-			return false, fmt.Errorf("determining existence of a ECR repository: %w", err)
-		}
+	ok, err := hasECRCreateDependenciesMet(state, meta.ApplicationDeclaration)
+	if err != nil {
+		return false, fmt.Errorf("checking ECR create dependencies: %w", err)
+	}
 
-		if !exists {
-			return false, nil
-		}
+	if !ok {
+		return false, nil
+	}
+
+	ok, err = hasIngressCreateDependencies(state, meta.ApplicationDeclaration)
+	if err != nil {
+		return false, fmt.Errorf("checking ingress create dependencies: %w", err)
+	}
+
+	if !ok {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+func hasECRCreateDependenciesMet(state *clientCore.StateHandlers, application v1alpha1.Application) (bool, error) {
+	if !application.Image.HasName() {
+		return true, nil
+	}
+
+	exists, err := state.ContainerRepository.ApplicationHasImage(application.Metadata.Name)
+	if err != nil {
+		return false, fmt.Errorf("determining existence of a ECR repository: %w", err)
+	}
+
+	return exists, nil
+}
+
+func hasIngressCreateDependencies(state *clientCore.StateHandlers, application v1alpha1.Application) (bool, error) {
```

Comment:
follow the same naming as the rest of these functions: `hasIngressCreateDependenciesMet()`?